Extended Project 3/lib/logic.rb

```diff
@@ -0,0 +1,58 @@
+# Esta linea está llamando a la clase Socket que tiene Ruby
+require 'socket'
+require_relative 'logic_methods'
+
+# Aquí está iniciando una conexión en el puerto 8000
+server = TCPServer.new 8000
+
+loop do
+  # Espera a que un cliente se conecte
+  client = server.accept
+  result = ""
+  code_http = "200"
+  content_type = "text/html"
+
+  # Recuperando toda la información enviada en la petición (request)
+  req = client.gets
+  # Cuando en algún momento la llamada sea por el cache del browser
+  # el valor de la solicitud (request) será nil
+  unless req == nil
+    # El formato del valor devuelto de req es
+    # GET data HTTP/1.1\r\n
+    # Usamos el split para obtener solo la parte de data
+    request = req.split(" ")[1]
+    if request.include?("/?member=")
+      request["/?member="] = ""
+      if check_file_data_existence(request)
+        data = read_string(request + ".txt").split("---------")
+        result = read_string("about.html")
+        8.times { |val| result["YYYYYYYY#{val + 1}"] = data[val]; }
```

Comment:
Could you explain me this part please?

Reply:
Reads if the requested name exists as a file, separates the input in 8 parts which are separated by "-"s, then reads the about.html and replaces the "Y"s with the text content. Which appears 8 times.